### PR TITLE
Add Nokogiri XML builders

### LINKED ISSUE #8: Stop using erbs to render xml

### PR DESCRIPTION
**Pull request by llimllib**
* Build XML with nokogiri instead of ERBs
* Test each request's XML for valid SOAP
* Add byebug back in if we're not in jruby

Fixes #8

### CODE REVIEW

**Comment by alex** [comment]
I'll let whoever is reviewing have the final say, but it might be easier to review if we did the "split erquests into their own files" in a seperate PR, so it's easier to see what changed?

**Comment by alex** [comment]
(Thanks for working on this! :sparkles: :cake: :sparkles:)

**Comment by alex** [comment]
this can be rebased now! thanks

File: src/vbms/requests/upload_document_with_associations.rb [hunk 1]
```diff
@@ -25,22 +25,39 @@ def template
         VBMS.load_erb('upload_document_xml_template.xml.erb')
       end
 
+      def received_date
+        @received_at.getlocal("-05:00").strftime("%Y-%m-%d-05:00")
+      end
+
+      # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
       def render_xml
-        # TODO: this is wrong
```
Comment (alex) [inline src/vbms/requests/upload_document_with_associations.rb]: This TODO got lost.

**Comment by alex** [comment]
I hate to ask, but what do you think about doing this as 1-request-per-PR so they're smaller.

File: src/vbms/requests/upload_document_with_associations.rb (hunk 1 above)
Comment (llimllib) [inline src/vbms/requests/upload_document_with_associations.rb]: I don't know what it meant, and it has been production code for >6 months now, so I deleted it intentionally

**Comment by llimllib** [comment]
I really dislike the idea of leaving the erb/noko machinery in there together?

File: src/vbms/requests/upload_document_with_associations.rb (hunk 1 above)
Comment (alex) [inline src/vbms/requests/upload_document_with_associations.rb]: Well, I have no idea what "external_id" is, but I'm skeptical all of these are supposed to be 123.

**Comment by alex** [comment]
@llimllib yeah, I think we should do one request at a time, totally remove the `erb` stuff and replace it with nokogiri

**Comment by harrisj** [comment]
I've read through this Pull Request and although it's a bunch of files, the commits are relatively straightforward in what they do. I think it is fine to merge. :+1:

**Pull request merged**